src/ui/units/ql/store/actions/ql.ts

```diff
@@ -1185,6 +1185,14 @@ export const updateChart = (data: QlConfig, mode?: EntryUpdateMode) => {
     };
 };
 
+export const updateChartAndDoAction = (preparedChartData: QlConfig, action: () => void) => {
+    return async function (dispatch: AppDispatch<QLAction>) {
+        await dispatch(updateChart(preparedChartData, EntryUpdateMode.Publish));
+
+        action();
```

Comment:
Why is it done this way? Is it possible to call action at the place of use with `then` or `wait`?

Reply:
I tried to use `updateChart` action with `await` and looked for more ways, but I didn't find anything, so I decided to put it in a separate action